README.md

```diff
@@ -12,27 +14,75 @@ When creating your main angular module, include `vokal.majora` in the list of in
 angular.module( "myApp", [ "vokal.majora" ] );
 ```
 Include `Majora` into any modules where you want to use the bitmask service, and use the following functions to manipulate bitmasks.
-###toggleBits( number, bitsArray, newValue )
-@param {Number} number - the number to be mutated  
-@param {Number} bitsArray - selects the bits to mutate in number  
-@param {Bool} newValue - new value of the bits selected by bitsArray
-
-Use: given an input `number`, update the bits matched by `bitsArray` to be `newValue`  
-Example: Say number is 8 (binary 1000) and you want to turn on the 1st (1 or 0001) and 3rd (4 or 0100)
-least significant bits, then bitsArray would be [ 1, 4 ] and newValue would be true. The returned value
+
+## <a name="section-interface"></a>Interface
+
+The following methods can be called on the `Majora` service once injected into your Angular code.
+
+* [toggleBits( number, bitsArray, newValue )](#method-toggleBits)
+* [testOptimistic( number, bitsArray )](#method-testOptimistic)
+* [testPessimistic( number, bitsArray )](#method-testPessimistic)
+
+### Methods
+
+#### <a name="method-toggleBits"></a>`toggleBits( number, bitsArray, newValue )`
+
+Given an input `number`, update the bits matched by `bitsArray` to be `newValue`
+
+##### Arguments
+
+1. `number`    | *Number* | the number to be mutated
+
+2. `bitsArray` | *Number* | selects the bits to mutate in number
+
+3. `newValue`  | *Bool*   | new value of the bits selected by bitsArray
+
+##### Response
+
+Returns a *Number*, which is the decimal representation of the `number` argument with the selected bits modified
```

Comment:
Header should be "Returns", and follow similar format as arguments.  That is,

    *Number* | the decimal representation of the `number` argument with the selected bits modified